tasks/request_from_archive/request_from_archive.py

```diff
@@ -132,7 +133,7 @@ def task(
         )
         retry_sleep_secs = DEFAULT_RESTORE_RETRY_SLEEP_SECS
 
-    # Get QUEUE URL
+    # Get QUEUE URLS
```

Comment:
Looks like the retrieval of the new queue url is occurring on line 598. Pull up.

Reply:
moving the url out of the function restore_object() will make it undefined.

Reply:
You'll need to pass it via parameter, same way we handle the status_update_queue_url.
As-is, the property is only retrieved the moment before it is needed, which means the function could process several files before realizing that the queue url is missing, causing a late failure.

Reply:
understood. Took some time to update unit tests but got that as argument now.